### DISCUSSION TITLE
Radio medium with modifiable data rate, transmission power and pathloss model?

### DISCUSSION THREAD

**Discussion by arahman33** [inet-framework/inet#935] [General]
I am currently using UnitDiskRadioMedium where I can set the data rate as desired. However, there is no concept of transmission power in UnitDiskRadioMedium. Moreover, I saw that the bitrate, even though initialized, has no effect on the actual transmission. So I changed the medium to IeeeScalarRadioMedium. Now I am able to set the transmission power, but I am no longer able to set the data rates (I am restricted by the opMode). 

My question is, is there any medium that allows me to set the data rate (without restrictions) as well as the transmission power? I would also like to set the path loss model to be used in the simulation.

Thanks

**Comment by levy** [comment]
In the latest INET, you can use ApskRadio and ScalarRadioMedium. In older versions these were called ApskScalarRadio and ApskScalarRadioMedium.
